pkg/dependencies/factory.go

```diff
@@ -252,6 +253,25 @@ func (f *Factory) WithGovc() *Factory {
 	return f
 }
 
+func (f *Factory) WithTink(clusterConfigFile string) *Factory {
+	f.WithExecutableBuilder()
+	tinkerbellDatacenterConfig, err := v1alpha1.GetTinkerbellDatacenterConfig(clusterConfigFile)
```

Comment:
nit: we could move this to inside the build step, that way we avoid calling it more than once. It also allows you to handle the error properly instead of just return nil and letting it explode